experimental/displaynames/tests/tests.rs

```diff
@@ -0,0 +1,54 @@
+use icu_displaynames::{DisplayNamesOptions, LocaleDisplayNamesFormatter};
+use icu_locid::locale;
+use icu_locid::Locale;
+use std::str::FromStr;
+
+#[test]
+fn test_concatenate() {
+    #[derive(Debug)]
+    struct TestCase<'a> {
+        pub input_1: &'a Locale,
+        pub expected: &'a str,
+    }
+    let cases = [
```

Comment:
Suggested cases:

- de-Latn-CH
- zh-Hans-CN
- es-419-fonipa

Also see if there are any cases in data where you have ambiguity between script and region, like if there is an entry for both "zh-Hant" and "zh-TW"

Reply:
Added the test cases.